performance_manager/lib/l0_rt_vehicle_positions.py

```diff
@@ -35,57 +31,89 @@ def get_vp_dataframe(to_load: Union[str, List[str]]) -> pandas.DataFrame:
         "vehicle_id",
     ]
 
-    return read_parquet(to_load, columns=vehicle_position_cols)
+    vehicle_position_filters = [
+        ("current_status", "!=", "None"),
+        ("current_stop_sequence", ">=", 0),
+        ("stop_id", "!=", "None"),
+        ("vehicle_timestamp", ">", 0),
+        ("direction_id", "in", (0, 1)),
+        ("route_id", "!=", "None"),
+        ("start_date", "!=", "None"),
+        ("start_time", "!=", "None"),
+        ("vehicle_id", "!=", "None"),
+    ]
+
+    result = read_parquet(
+        to_load, columns=vehicle_position_cols, filters=vehicle_position_filters
+    )
+
+    process_logger.add_metadata(row_count=result.shape[0])
+    process_logger.log_complete()
+
+    return result
 
 
-def transform_vp_dtyes(vehicle_positions: pandas.DataFrame) -> pandas.DataFrame:
+def transform_vp_datatypes(
+    vehicle_positions: pandas.DataFrame,
+) -> pandas.DataFrame:
     """
     ingest dataframe of vehicle position data from parquet file and transform
     column datatypes
     """
+    process_logger = ProcessLogger(
+        "vp.transform_datatypes", row_count=vehicle_positions.shape[0]
+    )
+    process_logger.log_start()
+
     # current_staus: 1 = MOVING, 0 = STOPPED_AT
     vehicle_positions["is_moving"] = numpy.where(
         vehicle_positions["current_status"] != "STOPPED_AT", True, False
     ).astype(numpy.bool8)
     vehicle_positions = vehicle_positions.drop(columns=["current_status"])
 
-    # store start_date as Int64 [nullable] instead of string
+    # store start_date as int64 instead of string
     vehicle_positions["start_date"] = pandas.to_numeric(
         vehicle_positions["start_date"]
-    ).astype("Int64")
+    ).astype("int64")
 
     # rename current_stop_sequence as stop_sequence
-    # and convert to Int64 [nullable]
+    # and convert to int64
     vehicle_positions.rename(
         columns={"current_stop_sequence": "stop_sequence"}, inplace=True
     )
     vehicle_positions["stop_sequence"] = pandas.to_numeric(
         vehicle_positions["stop_sequence"]
-    ).astype("Int64")
+    ).astype("int64")
 
-    # store direction_id as Int64 [nullable]
+    # store direction_id as bool
     vehicle_positions["direction_id"] = pandas.to_numeric(
         vehicle_positions["direction_id"]
-    ).astype("Int64")
+    ).astype(numpy.bool8)
 
-    # store start_time as seconds from start of day (Int64 [nullable])
+    # store start_time as seconds from start of day as int64
     vehicle_positions["start_time"] = (
         vehicle_positions["start_time"]
         .apply(start_time_to_seconds)
-        .astype("Int64")
+        .astype("int64")
     )
 
+    process_logger.log_complete()
     return vehicle_positions
 
 
-def join_gtfs_static(
+def join_vp_with_gtfs_static(
```

Comment:
renamed to avoid naming conflict with the same function in trip update processing.